hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractJsonGetter.java

```diff
@@ -0,0 +1,196 @@
+            String path = paths[i];
+            String arrayIndexText = getIndexTextOrNull(path);
+            if (arrayIndexText == null) {
+                // non array case
+
+
+                if (!findAttribute(parser, path)) {
+                    return null;
+                }
+            } else {
+                // array case
+                if ("any".equals(arrayIndexText)) {
+                    String lastPath = i + 1 < paths.length ? paths[i + 1] : null;
+                    return getMultiValue(parser, lastPath);
+                } else {
+                    JsonToken token = parser.getCurrentToken();
+                    if (token != JsonToken.START_ARRAY) {
+                        return null;
+                    }
+                    token = parser.nextToken();
+                    int arrayIndex = Integer.parseInt(arrayIndexText);
+                    for (int j = 0; j < arrayIndex; j++) {
+                        if (token == JsonToken.END_ARRAY) {
+                            return null;
+                        }
+                        parser.skipChildren();
+                        token = parser.nextToken();
+                    }
+                }
+            }
+        }
+        Object ret = convertJsonTokenToValue(parser);
+        parser.close();
+        return ret;
+    }
+
+    abstract JsonParser createParser(Object obj) throws IOException;
+
+    private boolean findAttribute(JsonParser parser, String path) throws IOException {
+        JsonToken token = parser.getCurrentToken();
+        if (token != JsonToken.START_OBJECT) {
+            return false;
+        }
+        parser.getCurrentToken();
+        while (true) {
+            token = parser.nextToken();
+            if (token == JsonToken.END_OBJECT) {
+                return false;
+            }
+            if (path.equals(parser.getCurrentName())) {
+                parser.nextToken();
+                return true;
+            } else {
+                parser.nextToken();
+                parser.skipChildren();
+            }
+        }
+    }
+
+    private MultiResult getMultiValue(JsonParser parser, String lastPath) throws IOException {
+        MultiResult<Object> multiResult = new MultiResult<Object>();
+
+        JsonToken currentToken = parser.currentToken();
+        if (currentToken != JsonToken.START_ARRAY) {
+            return null;
+        }
+        while (true) {
+            currentToken = parser.nextToken();
+            if (currentToken == JsonToken.END_ARRAY) {
+                break;
+            }
+            if (currentToken.isScalarValue()) {
+                multiResult.add(convertJsonTokenToValue(parser));
+            } else {
+                if (lastPath != null && findAttribute(parser, lastPath)) {
+                    multiResult.add(convertJsonTokenToValue(parser));
+                }
+                while (parser.getCurrentToken() != JsonToken.END_OBJECT) {
```

Comment:
This loops seems to fail when there are nested arrays as `new JsonGetter().getValue("{\"a\":[1,2,3,[3,4,5],4,5]}", "a[any]")` gives a `NullPointerException`